android/app/src/main/java/io/peggyjo/peggymeter/database/DataController.java

```diff
@@ -81,4 +88,12 @@ public MoodAdapter getMoodAdapter() {
     public SettingAdapter getSettingAdapter() {
         return settingAdapter;
     }
+
+    public void clearMoodListeners() {
```

Comment:
Could you make it synchronized to make sure we won't have concurrent exception?

Reply:
done